### PR TITLE
Autotools fixes

### PR DESCRIPTION
**Pull request by mjeanson**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,8 +20,8 @@ config.status
 configure
 coverage/
 coverage.info
-unit-test
-integration-test
+unit.test
+integration.test
```
[the remark this answers is not in the document]
Comment (mjeanson) [reply .gitignore:24]: Without using a TEST_EXTENSIONS I couldn't create a dependency between the log files, I got the following error when I ran autoreconf :

```
Makefile.am:45: warning: user target 'integration-test.log' defined here ...
/usr/share/automake-1.16/am/check2.am: ... overrides Automake target 'integration-test.log' defined here
autoreconf: automake failed with exit status: 1
```

I chose **.test** but that was arbitrary.

For the gritty details, you can have a look at https://www.gnu.org/software/automake/manual/html_node/Parallel-Test-Harness.html

[the remark this answers is not in the document]
Comment (freeekanayaka) [reply .gitignore:24]: Okay, that seems fine. However, the reason for introducing the dependency in the first place is to support parallel tests right? I'm wondering if there's a way to fix that instead.

[the remark this answers is not in the document]
Comment (mjeanson) [reply .gitignore:24]: Maybe, the problem seems to be binding to the same port in multiple tests. Running `make -j4 check` should trigger it.

[the remark this answers is not in the document]
Comment (freeekanayaka) [reply .gitignore:24]: > Maybe, the problem seems to be binding to the same port in multiple tests. Running `make -j2 check` should trigger it.

Okay, I'll see what I can do about it. There are other good fixes in this PR, would you mind just dropping the one that workarounds `make -j2 check`? I think that'd be the test binary rename and the log file Makefile dependency trick.

[the remark this answers is not in the document]
Comment (mjeanson) [reply .gitignore]: I removed **fix: tests failure when invoked with 'make -jX check'** from this PR.

**Pull request merged**